appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockContainer.java

```diff
@@ -256,7 +256,9 @@ protected boolean acceptableSource(DragSource source) {
     if (component instanceof MockVisibleComponent) {
       // Sprites are only allowed on Canvas, not other containers.
       // Map features are only allowed on Map, not other containers.
-      if (!(component instanceof MockSprite) && !(component instanceof MockMapFeature)) {
+      // Chart Data components are only allowed on Charts, not other containers.
+      if (!(component instanceof MockSprite) && !(component instanceof MockMapFeature)
```

Comment:
@ewpatton Should we refactor this to a separate method or to a Map?

Reply:
I would not worry about that for this PR. In theory it's possible, but GWT is also a bit weird with reflection and classes.

Reply:
Fair enough!